go/utils.go

```diff
@@ -157,3 +157,25 @@ func WaitForTask(taskRef xenapi.TaskRef, delay int) error {
 	}
 	return nil
 }
+
+func WaitForSRReady(srRef xenapi.SRRef) error {
+	token := ""
+	maxTries := 3
+	eventTypes := []string{"PBD"}
+	for i := 0; i < maxTries; i++ {
+		eventBatch, err := xenapi.Event.From(session, eventTypes, token, 10.0)
+		if err != nil {
+			return err
+		}
+		token = eventBatch.Token
+		for _, event := range eventBatch.Events {
+			if event.Class == "pbd"  {
+				pbdRecord := event.Snapshot.(map[string]interface{})
+			    if pbdRecord["SR"].(string) == string(srRef) && pbdRecord["currently_attached"].(bool) == true {
+					return nil
+				}
+			}
+		}
+	}
+	return nil
```

Comment:
Wouldn't this be an error when having the `maxTries` reached but no expected event received?

Reply:
Yes, an error will be better.